framework/PostProcessors/ValueDuration.py

```diff
@@ -0,0 +1,168 @@
+import numpy as np
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from .PostProcessor import PostProcessor
+from utils import utils
+from utils import InputData
+import Runners
+#Internal Modules End-----------------------------------------------------------
+
+class ValueDuration(PostProcessor):
+  """
+    Constructs a load duration curve.
+    x-axis is time spent above a particular variable's value,
+    y-axis is the value of the variable.
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    ## This will replace the lines above
+    inSpec= super(ValueDuration, cls).getInputSpecification()
+    inSpec.addSub(InputData.parameterInputFactory('target',
+                                                  contentType=InputData.StringListType,
+                                                  strictMode=True))
+    inSpec.addSub(InputData.parameterInputFactory('bins',
+                                                  contentType=InputData.IntegerType))
+    return inSpec
+
+  def __init__(self, messageHandler):
+    """
+      Constructor
+      @ In, messageHandler, MessageHandler, message handler object
+      @ Out, None
+    """
+    PostProcessor.__init__(self, messageHandler)
+    self.numBins = None # integer number of bins to use in creating the duration curve. TODO default?
+    self.dynamic = True
+
+  def _localReadMoreXML(self, xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.Element, Xml element node
+      @ Out, None
+    """
+    paramInput = self.getInputSpecification()()
+    paramInput.parseNode(xmlNode)
+    self._handleInput(paramInput)
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the parsed paramInput for this class.
+      @ In, paramInput, ParameterInput, the already-parsed input.
+      @ Out, None
+    """
+    for child in paramInput.subparts:
+      tag = child.getName()
+      if tag == 'target':
+        self.targets = set(child.value)
+      elif tag == 'bins':
+        self.numBins = child.value
+
+  def initialize(self, runInfo, inputs, initDict):
+    """
+      Method to initialize the pp.
+      In here the working dir is grepped.
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+      @ Out, None
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
```

Comment:
Do you need this method? Currently, it just call the parent class's method.

Reply:
I think I was using it, then ended up not doing so.  Removing.